common/common.h

```diff
@@ -8,6 +8,12 @@
 #include <vector>
 #include <sstream>
 
+#if defined(LLAMA_USE_CURL)
+#include <curl/curl.h>
+#include <curl/easy.h>
+#include <future>
+#endif
+
```

Comment:
We should keep `curl` contained in a single source: `common/common.cpp`. Move the `common_get_hf_file` to `common/common.cpp` and keep the includes in the `.cpp` file.